packages/atomic/src/components/commerce/atomic-commerce-interface/atomic-commerce-interface.spec.ts

```diff
@@ -0,0 +1,550 @@
+  const state: {language: string; query?: string} = {language: 'en'};
+  return {
+    ...originalModule,
+    buildContext: vi.fn(() => {
+      const context = {
+        state: {...state},
+        setLanguage: (language: string) => {
+          context.state.language = language;
+        },
+      };
+      return context;
+    }),
+
+    buildProductListing: vi.fn(() => ({
+      summary: vi.fn(() => ({
+        subscribe: vi.fn(() => ({
+          unsubscribe: vi.fn(),
+        })),
+      })),
+      urlManager: vi.fn(() => ({
+        subscribe: vi.fn(() => ({
+          unsubscribe: vi.fn(),
+        })),
+      })),
+      executeFirstRequest: vi.fn(),
+    })),
+    buildSearch: vi.fn(() => {
+      const mockState = {firstRequestExecuted: true};
+      const mockSubscribe = vi.fn((callback) => {
+        return callback;
+      });
+      return {
+        summary: vi.fn(() => ({subscribe: mockSubscribe, state: mockState})),
+        urlManager: vi.fn(() => ({
+          subscribe: vi.fn(() => ({unsubscribe: vi.fn()})),
+        })),
+        executeFirstSearch: vi.fn(),
+      };
+    }),
+    loadQueryActions: vi.fn(() => ({
+      updateQuery: vi.fn(({query}) => ({
+        type: 'updateQuery',
+        payload: {query},
+      })),
+    })),
+    buildCommerceEngine: vi.fn(
+      (config: {configuration: CommerceEngineConfiguration}) => {
+        if (!config.configuration.organizationId) {
+          throw new Error('Invalid configuration: organizationId is required');
+        }
+        return {
+          [stateKey]: state,
+          ...config,
+          dispatch: vi.fn(),
+          addReducers: vi.fn(),
+          disableAnalytics: vi.fn(),
+          enableAnalytics: vi.fn(),
+          logger: {
+            info: vi.fn(),
+            warn: vi.fn(),
+            error: vi.fn(),
+          },
+          subscribe: vi.fn(() => {
+            return {
+              unsubscribe: vi.fn(),
+            };
+          }),
+        };
+      }
+    ),
+  };
+});
+
+@customElement('test-element')
+@bindings()
+export class TestElement
+  extends InitializeBindingsMixin(LitElement)
+  implements InitializableComponent<CommerceBindings>
+{
+  @state()
+  public bindings: CommerceBindings = {} as CommerceBindings;
+  @state() public error!: Error;
+
+  public addStyles(styles: string | CSSStyleSheet) {
+    if (typeof styles === 'string') {
+      const styleSheet = new CSSStyleSheet();
+      styleSheet.replaceSync(styles);
+      this.bindings.addAdoptedStyleSheets(styleSheet);
+    } else if (styles instanceof CSSStyleSheet) {
+      this.bindings.addAdoptedStyleSheets(styles);
+    }
+  }
+
+  public initialized = false;
+
+  public render() {
+    return html`test-element`;
+  }
+
+  initialize = vi.fn();
+}
+
+const commerceEngineConfig: CommerceEngineConfiguration =
+  getSampleCommerceEngineConfiguration();
+describe('AtomicCommerceInterface', () => {
```

Comment:
In general I personally like organizing tests by public method:

e.g.,

```typescript
describe('AtomicCommerceInterface', () => {
  it('should ...', () => {});
  describe('#initialize', () => {
    it('should ...', () => {});
    it('when ... should ...', ,() => {});
    describe('when ...', () => {
      it('should ...', () => {});
      it('should ...', () => {});
    });
  })
});
```

That being said, we need to come to some kind of concensus within the team before we formalize this in any way.

Consider this a comment, not a request for change ;)